streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -107,7 +108,7 @@ public StreamTask(TaskId id,
                       ThreadCache cache,
                       Time time,
                       final RecordCollector recordCollector) {
-        super(id, applicationId, partitions, topology, consumer, restoreConsumer, false, stateDirectory, cache);
+        super(id, applicationId, partitions, topology, consumer, restoreConsumer, false, stateDirectory, cache, time, config.getLong(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG));
```

Comment:
If we are always writing the checkpointing file upon committing, do we still need this parameter? Or we can just execute the logic of `Checkpointer.checkpoint()` without the timing conditional?